search/includes/classes/queue/class-cron.php

```diff
@@ -212,29 +212,27 @@ public function sweep_jobs() {
 
 		$this->queue->free_deadlocked_jobs();
 
-		// If job limit has been surpassed, don't schedule any new jobs
-		if ( $this->get_processor_job_count() > self::MAX_PROCESSOR_JOB_COUNT ) {
-			return;
-		}
+		$job_count = $this->get_processor_job_count();
+		while ( ! is_wp_error( $job_count ) && $job_count < self::MAX_PROCESSOR_JOB_COUNT ) {
 
-		$i = 1; // Start incrementor at 1 since the first loop is the first job
-		do {
-			// Prevent loop from scheduling more than the maximum allowed jobs in any single iteration
-			if ( $i > self::MAX_PROCESSOR_JOB_COUNT ) {
-				return;
+			$schedule_success = $this->schedule_batch_job();
+			if ( is_wp_error( $schedule_success ) || ! $schedule_success ) {
+				break;
 			}
-			
-			$this->schedule_batch_job();
 
-			$i++;
-		} while ( $this->get_processor_job_count() < self::MAX_PROCESSOR_JOB_COUNT );
-		// Prevent continuing if the current processor job count goes over the max allowed job count
+			$job_count = $this->get_processor_job_count();
```

Comment:
Would $job_count++ work here instead of a new query?

Reply:
I thought about it and it probably would. But it would introduce a bigger change of race condition issue. This cron is not the only place the jobs is added so I figure it is safer to keep the query in as it was there before.

(The previous code used a local $i counter for some reason alongside with the `$this->get_processor_job_count() < self::MAX_PROCESSOR_JOB_COUNT` so it would also do the query on each iteration.)